osquery/tables/system/windows/userassist.cpp

```diff
@@ -0,0 +1,179 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed in accordance with the terms specified in
+ *  the LICENSE file found in the root directory of this source tree.
+ */
+
+#include <osquery/core.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+#include <osquery/tables/system/windows/registry.h>
+#include <osquery/tables/system/windows/userassist.h>
+#include <osquery/utils/conversions/split.h>
+#include <osquery/utils/conversions/tryto.h>
+#include <osquery/utils/system/time.h>
+#include <string>
+
+namespace osquery {
+namespace tables {
+
+constexpr auto kFullRegPath =
+    "\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Explorer\\UserAssist";
+
+// Decode ROT13 sub key value
+std::string rot_decode(std::string& value_key_reg) {
+  std::string decoded_value_key;
+
+  for (std::size_t i = 0; i < value_key_reg.size(); i++) {
+    if (isalpha(value_key_reg[i])) {
+      if (value_key_reg[i] >= 'a' && value_key_reg[i] <= 'm') {
+        decoded_value_key.append(1, value_key_reg[i] + 13);
+      } else if (value_key_reg[i] >= 'm' && value_key_reg[i] <= 'z') {
+        decoded_value_key.append(1, value_key_reg[i] - 13);
+      } else if (value_key_reg[i] >= 'A' && value_key_reg[i] <= 'M') {
+        decoded_value_key.append(1, value_key_reg[i] + 13);
+      } else if (value_key_reg[i] >= 'M' && value_key_reg[i] <= 'Z') {
+        decoded_value_key.append(1, value_key_reg[i] - 13);
+      }
+    } else {
+      decoded_value_key.append(1, value_key_reg[i]);
+    }
+  }
+  return decoded_value_key;
+}
+
+// Get last exeution time
+auto last_execute(std::string& assist_data) {
+  std::string last_run_string = assist_data.substr(120, 16);
+
+  // If timestamp is zero dont convert to UNIX Time
+  if (last_run_string == "0000000000000000") {
+    std::string no_timestamp = "";
+    return no_timestamp;
+  } else {
+    // swap endianess
+    std::reverse(last_run_string.begin(), last_run_string.end());
+
+    char temp;
+    for (std::size_t i = 0; i < last_run_string.length(); i += 2) {
+      temp = last_run_string[i];
+      last_run_string[i] = last_run_string[i + 1];
```

Comment:
If `last_run_string` size is 1 then this will overflow.

Reply:
The check for `assist_data` also makes sure that `last_run_string` will be size 16